x-pack/legacy/plugins/reporting/export_types/csv/server/execute_job.ts

```diff
@@ -6,18 +6,24 @@
 
 import Hapi from 'hapi';
 import { i18n } from '@kbn/i18n';
-import { ElasticsearchServiceSetup, KibanaRequest } from '../../../../../../../src/core/server';
+import {
+  ElasticsearchServiceSetup,
+  KibanaRequest,
+  IUiSettingsClient,
+} from '../../../../../../../src/core/server';
 import { CSV_JOB_TYPE } from '../../../common/constants';
 import { cryptoFactory } from '../../../server/lib';
 import { ESQueueWorkerExecuteFn, ExecuteJobFactory, Logger, ServerFacade } from '../../../types';
 import { JobDocPayloadDiscoverCsv } from '../types';
 import { fieldFormatMapFactory } from './lib/field_format_map';
 import { createGenerateCsv } from './lib/generate_csv';
 import { getFieldFormats } from '../../../server/services';
+import { ReportingPlugin } from '../../../server/plugin';
 
 export const executeJobFactory: ExecuteJobFactory<ESQueueWorkerExecuteFn<
   JobDocPayloadDiscoverCsv
->> = function executeJobFactoryFn(
+>> = async function executeJobFactoryFn(
+  reporting: ReportingPlugin,
   server: ServerFacade,
   elasticsearch: ElasticsearchServiceSetup,
   parentLogger: Logger
```

Comment:
Minor: now that you introduced this `ReportingPlugin` internal that currently contains SO accessors such as `reporting.getSavedObjectsClient(fakeRequest)`, it would probably makes sense to move the `elasticsearch` accessors inside it too instead of passing it as an additional parameter, for consistency.

Reply:
the thought occurred to me as well, so 👍

Reply:
I'd like to take this on in a future PR. It would make sense to get this in before working on the migration of our "config" usage on the server. Config can also follow the pattern of using an accessor on the reporting object.

Reply:
fine with me